### PR TITLE
Teach deployment HPA describer about new HPA

### PR DESCRIPTION
**Pull request by DirectXMan12**
The deployment HPA controller now knows about all types of metric
specifications, and handles when a single HPA has multiple different
metric specs, printing each on it's own line.

### CODE REVIEW

**Comment by DirectXMan12** [comment]
cc @ncdc

**Comment by DirectXMan12** [comment]
New output:

### Single metric ###
```
Autoscaling:	between 1 and 10 replicas targetting 80% cpu average per pod
```

### Multiple metrics ###
```
Autoscaling:	between 1 and 10 replicas
		  targetting 256Mi memory average per pod
		  targetting 1Ki bytes-per-second average per pod
		  targetting 500 hits-per-second on Service frontend
		  targetting 80% cpu average per pod
```

**Comment by DirectXMan12** [comment]
NB: I've changed the wording here away from using the phrasing "triggered at", because that's a bit of a misnomer.  The HPA is *not* threshold-based.  Rather, it "aims" for a particular load, quasi-constantly scaling the replica count by the ratio of current to target.

File: pkg/cmd/cli/describe/deployments.go [hunk 1]
```diff
@@ -321,21 +321,56 @@ func printAutoscalingInfo(res []schema.GroupResource, namespace, name string, kc
...
-		// TODO(rebase-1.6): flesh this out for full hpa support, consider using upstream code
-		// For right now I'm just doing parity with 1.5
-		for _, metric := range hpa.Spec.Metrics {
-			if metric.Type == autoscaling.ResourceMetricSourceType && metric.Resource.Name == kapi.ResourceCPU && metric.Resource.TargetAverageUtilization != nil {
-				cpuUtil = fmt.Sprintf(", triggered at %d%% CPU usage", *metric.Resource.TargetAverageUtilization)
+		fmt.Fprintf(w, "Autoscaling:\tbetween %d and %d replicas", *hpa.Spec.MinReplicas, hpa.Spec.MaxReplicas)
+
+		targetDescriptions := formatHPATargets(&hpa)
+		if len(targetDescriptions) == 1 {
+			fmt.Fprintf(w, " targetting %s\n", targetDescriptions[0])
```
Comment (ncdc) [inline pkg/cmd/cli/describe/deployments.go]: targeting

File: pkg/cmd/cli/describe/deployments.go [hunk 2]
```diff
@@ -321,21 +321,56 @@ func printAutoscalingInfo(res []schema.GroupResource, namespace, name string, kc
...
 			}
 		}
-		fmt.Fprintf(w, "Autoscaling:\tbetween %d and %d replicas%s\n", *hpa.Spec.MinReplicas, hpa.Spec.MaxReplicas, cpuUtil)
 		// TODO: Print a warning in case of multiple hpas.
 		// Related oc status PR: https://github.com/openshift/origin/pull/7799
 		break
 	}
 }
 
+// printHPATargets formats a list of HPA targets in human readable form.  It functions similarly to the
```
Comment (ncdc) [inline pkg/cmd/cli/describe/deployments.go]: formatHPATargets

**Pull request merged**